### PR TITLE
Homework week2

### PR DESCRIPTION
**Pull request by Bogdy87**
Hi,

It was difficult the last part (optional), and I have tried to solve it, searching from the internet, as I couldn't do it alone. I hope it is ok but difficult to understand.

### CODE REVIEW

File: javascript/javascript1/week2/Homework.js
Comment (jason-vasilev) [inline javascript/javascript1/week2/Homework.js:12]: I know the assignment hasn't written it correctly, but please remember to use `cammelCase`. The function would be written as `getFullName` and the first property as `firstName`.
If you install [Code Spell Checker](https://marketplace.visualstudio.com/items?itemName=streetsidesoftware.code-spell-checker), you would get squiggly blue underlines and warning for "Unknown word", which will disappear once you correct it. It checks for proper spelling not just code comments but also variable, function, and object names.

Screenshot 2024-05-04 at 7 25 01 AM

File: javascript/javascript1/week2/Homework.js
Comment (jason-vasilev) [inline javascript/javascript1/week2/Homework.js:16]: The `else` section automatically will execute if the `if` condition is not met. No need to explicitly set it as `else if (gender === "female") {`. Just `} else {` will suffice here.

File: javascript/javascript1/week2/Homework.js
Comment (jason-vasilev) [inline javascript/javascript1/week2/Homework.js:15]: You've repeated `${firstName} ${surname}` three times in this function. There is a principle called [DRY (Don't Repeat Yourself)](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself) which recommends to make variables and functions for code that you write more than once. Eg.:
```
const fullName = `${firstName} ${surname}`;
```

File: javascript/javascript1/week2/Homework.js
Comment (jason-vasilev) [inline javascript/javascript1/week2/Homework.js:26]: Good that you run this check. It will be bad data to show only "Lady " and no name. May be you could add check in the `if else` logic for minimum allowed `fullName.length`?
One way to do it would be to use the [`trim` method for removing any empty space](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/trim) before and after strings.

Example: 
```
const fullName = `${firstname.trim()} ${surname.trim()}`;
```
And I will leave it to you to make the `if else` logic

File: javascript/javascript1/week2/Homework.js
Comment (jason-vasilev) [inline javascript/javascript1/week2/Homework.js:10]: Remember to delete old experiments. From one part to avoid confusion, and from second - to keep file size small.

File: javascript/javascript1/week2/Homework.js
Comment (jason-vasilev) [inline javascript/javascript1/week2/Homework.js:32]: The assignment requires that `You should get the today's day from the system.`. As a hint, I can direct you to look up `new Date()`.
Otherwise the rest of the code is perfect.

File: javascript/javascript1/week2/Homework.js
Comment (jason-vasilev) [inline javascript/javascript1/week2/Homework.js:44]: This will never execute as you use the AND operator `&&` and expect the same variable (temperature) to be above and bellow 15. You can test it by hardcoding temperature in your function call `const clothesToWear = howIsWeather(15)` and then trying to go to 16, 14 and seeing which message is logged.

File: javascript/javascript1/week2/Homework.js
Comment (jason-vasilev) [inline javascript/javascript1/week2/Homework.js:51]: Well done. If I had to do it, I wouldn't have come up with such a nifty solution. You are good with the numbers and super nice that you've added a comment explaining what's going on.
Only thing you can improve on this line is to change `let` for `const`.

File: javascript/javascript1/week2/Homework.js
Comment (jason-vasilev) [inline javascript/javascript1/week2/Homework.js:60]: I like what you've done in this function. Very tidy and easy to read. Also good code comments.

File: javascript/javascript1/week2/Homework.js
Comment (jason-vasilev) [inline javascript/javascript1/week2/Homework.js:103]: Very well testing again for "Queen". You can also add a test for the case with empty string.

File: javascript/javascript1/week2/Homework.js
Comment (jason-vasilev) [inline javascript/javascript1/week2/Homework.js:137]: Were you trying to do it with `for loop` initially?

**Review by jason-vasilev — APPROVED** [review]
The code works well. The comments that I've left you are just recommendations and small improvements. I'm approving now, so you are welcome to merge or apply some of the comments ( :

Please, remember to use Prettier. It makes it easier to read the code and see potential errors. Here you can read a guide how to set it up to automatically format on file save - https://www.digitalocean.com/community/tutorials/how-to-format-code-with-prettier-in-visual-studio-code

I am glad you've used the PR description field. Generally it's recommended to describe what is the problem the PR solves and how. Many companies use a set template for this, as first quick summary section, then instructions how to test, links to design sources or JIRA ticket (project management system).

It's alright to look up solutions, as long as you don't copy them completely. We are also here to help, so you can try reaching out to us for guidance.

Last but not least, remember to commit more often ;) It's a good practice for those cases when you might want to go back and cherry pick a particular code through a commit.

File: javascript/javascript1/week2/Homework.js
Comment (Bogdy87) [reply javascript/javascript1/week2/Homework.js:12]: I have installed now. Thanks.

File: javascript/javascript1/week2/Homework.js
Comment (Bogdy87) [reply javascript/javascript1/week2/Homework.js:103]: I have tested and always the number of students from the class  is always 0 and i don't know why.

**Pull request merged**
